### PR TITLE
prune expired proposals

### PR DESCRIPTION
**Pull request by nmarley**
Note: Since the number of votes required to delete a proposal was increased to 2/3 of the active masternodes, I am unable to fully test this on testnet. The delete votes seem to be valid and propagating to other nodes.

### CODE REVIEW

File: lib/models.py [hunk 1]
```diff
@@ -355,7 +355,7 @@ def approved_and_ranked(self, proposal_quorum, next_superblock_max_budget):
         return ranked
 
     @classmethod
-    def expired(self, dashd):
+    def expired(self):
```
Comment (tgflynn) [inline lib/models.py]: Doesn't this change require also changing how the method is called ?

Comment (nmarley) [reply lib/models.py]: It doesn't use the `dashd` argument and wasn't called with it initially, the method signature was the only thing wrong with it, so cleaning this up fixed the issue. It was erroring with some message about a missing positional argument before this change.

Comment (nmarley) [reply lib/models.py]: The call to it is here for reference: https://github.com/dashpay/sentinel/pull/16/files#diff-58a68114486f7532a706cd1aac030594R65

**Comment by tgflynn** [comment]
I think it would be better to add some additional delay after end_epoch before deleting.

I'm afraid that deleting immediately after end_epoch could lead to (rare) edge cases where there is competition between whether a proposal should be included in a superblock trigger or deleted.

**Comment by UdjinM6** [comment]
@tgflynn is there any validation for `end_epoch`? If not, it doesn't matter if we have delay or not, there always will be some probability of an edge case.

**Comment by tgflynn** [comment]
@UdjinM6 I don't follow you. Can you explain why there would be an edge case if deletion can only occur a certain time after end_epoch ?

Also what do you mean by "validation for end_epoch" ?

**Comment by tgflynn** [comment]
I mean the delay should be sufficient to avoid any problems with clock skew.  I would say at least 24 hours.

**Comment by UdjinM6** [comment]
My concerns were addressed in slack discussion, I agree that some delay is need (and should be enough in current design).

**Comment by nmarley** [comment]
To recap Slack discussion, we need to ensure that fudge window for proposal inclusion is honored, as well as waiting a longer time such as 1/2 the superblock window AFTER the end_epoch, to ensure that proposals aren't deleted prematurely.

**Comment by nmarley** [comment]
Waiting to add the delay before merging, should be another commit within the next few hours/couple days.

File: lib/constants.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# for constants which need to be accessed by various parts of Sentinel
+
+# Sentinel version
+VERSION = '1.0.1'
```
Comment (UdjinM6) [inline lib/constants.py]: This doesn't seem to be neither related to this specific PR nor used anywhere

File: requirements.txt [hunk 3]
```diff
@@ -1,7 +1,7 @@
 inflection==0.3.1
 peewee==2.8.3
 py==1.4.31
-pycodestyle==2.2.0
+pycodestyle==2.3.1
```
Comment (UdjinM6) [inline requirements.txt]: seems unrelated

File: .travis.yml [hunk 4]
```diff
@@ -13,4 +13,4 @@ script:
   - py.test -svv test/unit/
 
   # style guide check
-  - find ./lib ./test ./bin -name \*.py -exec pycodestyle --show-source --ignore=E501,E402 {} +
+  - find ./lib ./test ./bin -name \*.py -exec pycodestyle --show-source --ignore=E501,E402,E722 {} +
```
Comment (UdjinM6) [inline .travis.yml]: seems unrelated

**Review by UdjinM6 — COMMENTED** [review]
There are few unrelated changes which I think shouldn't be a part of this PR but otherwise looks good imo.

File: requirements.txt (hunk 3 above)
Comment (nmarley) [reply requirements.txt]: Correct, I will make this a separate PR

File: lib/constants.py (hunk 2 above)
Comment (nmarley) [reply lib/constants.py]: Removed from this branch, will add in a different PR
